src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/CSharpConvertBinder.cs

```diff
@@ -66,9 +71,45 @@ public void PopulateSymbolTableWithName(Type callingType, ArgumentObject[] argum
             base(type, conversionKind == CSharpConversionKind.ExplicitConversion)
         {
             ConversionKind = conversionKind;
+            _callingContext = callingContext;
             _binder = new RuntimeBinder(callingContext, isChecked);
         }
 
+        public int BinderEqivalenceHash
+        {
+            get
+            {
+                int hash = _callingContext?.GetHashCode() ?? 0;
+                hash = HashHelpers.Combine(hash, (int)ConversionKind);
+                if (IsChecked)
+                {
+                    hash = HashHelpers.Combine(hash, 1);
+                }
```

Comment:
hash = HashHelpers.Combine(hash, (int)IsChecked);

Reply:
IsChecked is bool. It would not convert to int. Not in c#

Reply:
Ah, my bad. Misread it :(